files/.github/workflows/ci.yml

```diff
@@ -18,32 +18,40 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+      <% if (pnpm) {%>- name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7<%}%>
       - name: Install Node
         uses: actions/setup-node@v3
         with:
           node-version: 14.x
-          cache: <%= yarn ? 'yarn' : 'npm' %>
+          cache: <%= pnpm ? 'pnpm' : yarn ? 'yarn' : 'npm' %>
       - name: Install Dependencies
-        run: <%= yarn ? 'yarn install --frozen-lockfile' : 'npm ci' %>
+        run: <%= pnpm ? 'pnpm install --frozen-lockfile' : yarn ? 'yarn install --frozen-lockfile' : 'npm ci' %>
       - name: Lint
-        run: <%= yarn ? 'yarn' : 'npm run' %> lint
+        run: <%= pnpm ? 'pnpm' : yarn ? 'yarn' : 'npm run' %> lint
       - name: Run Tests
-        run: <%= yarn ? 'yarn' : 'npm run' %> test
+        run: <%= pnpm ? 'pnpm' : yarn ? 'yarn' : 'npm run' %> test
 
   floating:
     name: "Floating Dependencies"
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v3
+      <% if (pnpm) {%>- name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7<%}%>
       - uses: actions/setup-node@v3
         with:
           node-version: 14.x
-          cache: <%= yarn ? 'yarn' : 'npm' %>
+          cache: <%= pnpm ? 'pnpm' : yarn ? 'yarn' : 'npm' %>
       - name: Install Dependencies
-        run: <%= yarn ? 'yarn install --no-lockfile' : 'npm install --no-shrinkwrap' %>
+        run: <%= pnpm ? 'pnpm install --no-lockfile' : yarn ? 'yarn install --no-lockfile' : 'npm install --no-shrinkwrap' %>
```

Comment:
I'm not sure `pnpm install` has `--no-lockfile` option

Reply:
I wasn't either as you don't find it on their docs page, but it actually has, and it works the same as yarn's.

```
pnpm install --help
// ...
// --no-lockfile                     Don't read or generate a `pnpm-lock.yaml` file
// ...
```

Reply:
@LevelbossMike good to know, thanks for confirming!